src/components/startup-form.tsx

```diff
@@ -0,0 +1,102 @@
+"use client"
+
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Button } from "@/components/ui/button";
+import { Form} from "@/components/ui/form";
+import { database } from "@/app/firebaseConfig";
+import { ref as dbRef, get, push, set } from "firebase/database";
+import { toast } from "sonner";
+import { useRouter } from "next/navigation";
+import { useEffect } from "react";
+import { StartupFormValues, startupSchema } from "@/shared/api/validations/startupSchema";
+import { TextField } from "./text-field";
+
+type StartupFormProps = {
+    startupId?: string;
+};
+
+export default function StartupForm({ startupId }: StartupFormProps) {
+    const router = useRouter();
+
+    const form = useForm<StartupFormValues>({
+        resolver: zodResolver(startupSchema),
+        defaultValues: {
+            year: "",
+            name: "",
+            description: "",
+            sector: "",
+            quote: "",
+            author: "",
+        },
+    });
+    useEffect(() => {
+        if (startupId) {
+            const startupRef = dbRef(database, `startups/${startupId}`);
```

Comment:
Could you please move this to the startupsService?

This way, you can centralize the GET, POST, and UPDATE methods. Also, if there's a change in the database name—like if we need to change 'startups' to 'emprendimientos'—you'll only need to update it in one place, instead of in every file.